### PR TITLE
ReadWriteMemory benchmark + optimize batch openings

### PR DESCRIPTION
**Pull request by moodlezoup**

### CODE REVIEW

File: jolt-core/src/jolt/vm/read_write_memory.rs [hunk 1]
```diff
@@ -270,16 +274,19 @@ where
...
+    #[tracing::instrument(skip_all, name = "MemoryReadWriteOpenings::open")]
     fn open(polynomials: &ReadWriteMemory<F, G>, opening_point: &Vec<F>) -> Self::Openings {
+        let chis = EqPolynomial::new(opening_point.to_vec()).evals();
         [
-            polynomials.a_read_write.evaluate(&opening_point),
-            polynomials.v_read.evaluate(&opening_point),
-            polynomials.v_write.evaluate(&opening_point),
-            polynomials.t_read.evaluate(&opening_point),
-            polynomials.t_write.evaluate(&opening_point),
+            polynomials.a_read_write.evaluate_at_chi(&chis),
```
Comment (sragss) [inline jolt-core/src/jolt/vm/read_write_memory.rs:281]: Parallelize these openings

File: jolt-core/src/jolt/vm/read_write_memory.rs [hunk 2]
```diff
@@ -355,14 +362,17 @@ where
...
     type Openings = [F; 3];
 
+    #[tracing::instrument(skip_all, name = "MemoryInitFinalOpenings::open")]
     fn open(polynomials: &ReadWriteMemory<F, G>, opening_point: &Vec<F>) -> Self::Openings {
+        let chis = EqPolynomial::new(opening_point.to_vec()).evals();
         [
-            polynomials.v_init.evaluate(&opening_point),
-            polynomials.v_final.evaluate(&opening_point),
-            polynomials.t_final.evaluate(&opening_point),
+            polynomials.v_init.evaluate_at_chi(&chis),
```
Comment (sragss) [inline jolt-core/src/jolt/vm/read_write_memory.rs:369]: Same here: Parallelize

**Review by sragss — APPROVED** [review]
Let's parallelize those eval points. Otherwise LGTM

**Comment by moodlezoup** [comment]
going to move these changes into a different PR

**Pull request closed without merging**